### ISSUE TITLE
Sentry integration and log audit for Python services

### ISSUE THREAD

**Issue by alexzurbonsen** [magma/magma#10507]
This issue serves as an umbrella issue for work related to embedding Sentry and auditing logs and exception handling in AGW Python code.

## Sentry
Sentry is currently configurable separately for each service. For a more detailed motivation and explanation of this configuration see #9863. Here, the focus is on a brief summary of the current status.

Three Sentry configuration options are available for each service: `send_all_errors`, `send_selected_errors`, `disabled`
While the first and the last option are self-explanatory, some more info on `send_selected_errors` is helpful. The intention is to introduce features to hand-pick errors, that are forwarded to Sentry, for example to exclude high volume or/and uninformative errors. Technically this is done by using the [`before_send`](https://docs.sentry.io/platforms/python/guides/aws-lambda/configuration/filtering/#using-beforesend) hook of Sentry’s init function and the logging parameter `extra` to flag selected errors. Furthermore a decorator is added to all gRPC endpoints of a service in order to include unexpected gRPC errors.
The `send_selected_errors` option has to be implemented for every service to be useable (otherwise it is equivalent to `disabled`).

Currently it has been implemented for
- magmad (#10349)
- directoryd (#10113)
- state (#10125)
- smsd (#10099)
- ctraced (#10110) 
- eventd (#10100)
- pipelined (#10374)
- subscriberdb (#10373)
- monitord (#10101)

Another configuration option is a client-side outbound filter that excludes certain message patterns from being sent to Sentry (#10276).

## Log and exception handling audit

The Sentry integration described above is accompanied by a complementary effort of auditing logs and exception handling in the AGW Python code. The goal is to obtain cleaner and clearer logs and enable useful Sentry integration. See #10027 for more details.
There, e.g. statistics about the number of logging and exception handling statements per service, can be found.

As part of these efforts tickets for most frequent issues on Sentry were created. Many of them are related to technical errors or timeouts in the communication between two services. Most of them require in-depth domain knowledge to figure out a solution. Thus, we tag the respective codeowners here as well.

### pipelined 
@magma/approvers-agw-pipelined 

- [x] #10407
- [x] #10409
- [x] #10708
- [x] #10785
- [x] #10961
- [x] #10983
- [x] #10984
- [x] #10420
- [x] #10342 
- [x] #10343 
- [ ] [added recently] #11485
- [ ] [added recently] #11708 

### magmad
@magma/approvers-agw-python

- [x] #10408
- [x] #10418

### subscriberdb
@magma/approvers-agw-python

- [x] #10387
- [x] #10782
- [x] [added recently] #11581 

### mobilityd
@magma/approvers-agw-python

- [x] #10846
- [x] #11097
- [x] #11106
- [x] #10419

### policydb
@magma/approvers-agw-python

- [x] #10783
- [x] #10964

### state

- [x] #10982 

### monitord
@magma/approvers-agw-python

- [x] #10163
- [ ] [added recently] #11705 

### eventd

- [ ] [added recently] #11576 

## Concept for definition of error / warning / info

tl,dr: platform and application errors need to be distinguished. This effects the usage of logs and sentry.

Both lines of work above draw on a definition of error, warning and info. Here, we summarise the current understanding, including definitions and usage.

The principal source of ambiguity is due to different usages of the term error. For magma developers an error is typically a bug, that is something unexpected that requires intervention on the code. For magma users on the other hand, an error is not a bug, but a failure of the working program. These kind of application errors may, for example, be due to faulty configurations. It is not necessarily unexpected but the user has to take action to get the software into a functioning state.

In conclusion:

- platform errors ~ bugs
- application error ~ user errors (configuration, broken communication etc.)

The two are differentiated by asking: who has to take action? The magma developers (platform error) or magma users (application error).

Both errors should be treated differently with regard to Sentry. Platform errors should usually be sent to Sentry but application errors should only be visible in the logs. If platform errors are high volume, it can also make sense to exclude them from being sent to Sentry and track them with an issue on GitHub instead.

Exclusion from Sentry can be achieved in different ways. One possibility is the opt-in mode `send_selected_errors` described above. For more details see https://github.com/magma/magma/issues/10027#issuecomment-965544441.

**Comment by sebathomas** [comment]
Closing this issue. While Sentry-related issues can still be found using the label [`sentry-cleanup-python`](https://github.com/magma/magma/labels/sentry-cleanup-python), most of the frequent ones have been excluded from Sentry or fixed, and the documentation of Sentry and different types of errors can now be found on:
https://github.com/magma/magma/wiki/Error-monitoring-and-triaging-workflow
